rs/consensus/utils/src/lib.rs

```diff
@@ -283,11 +291,14 @@ pub fn find_lowest_ranked_proposals(pool: &PoolReader<'_>, h: Height) -> Vec<Blo
         .fold(
             (None, Vec::new()),
             |(mut best_rank, mut best_proposals), proposal| {
-                if best_rank.is_none() || best_rank.unwrap() > proposal.rank() {
-                    best_rank = Some(proposal.rank());
-                    best_proposals = vec![proposal];
-                } else if Some(proposal.rank()) == best_rank {
-                    best_proposals.push(proposal);
+                // Only consider non-equivocating blockmakers
+                if !disqualified.contains(&proposal.signature.signer) {
+                    if best_rank.is_none() || best_rank.unwrap() > proposal.rank() {
```

Comment:
nit: alternatively (seeing unwraps makes me a bit uneasy :D)

```suggestion
                    if !best_rank.is_some_and(|rank| rank <= proposal.rank()) {
```

Reply:
done